inference-engine/src/mkldnn_plugin/utils/node_dumper.cpp

```diff
@@ -0,0 +1,203 @@
+
+    return;
+}
+
+void NodeDumper::dumpOutputBlobs(const MKLDNNNodePtr& node) const {
+    if (!shouldBeDumped(node))
+        return;
+
+    auto exec_order = std::to_string(node->getExecIndex());
+    auto nodeName = node->getName();
+    std::replace(nodeName.begin(), nodeName.end(), '\\', '_');
+    std::replace(nodeName.begin(), nodeName.end(), '/', '_');
+    std::replace(nodeName.begin(), nodeName.end(), ' ', '_');
+    std::replace(nodeName.begin(), nodeName.end(), ':', '-');
+
+    auto num_ports = node->getSelectedPrimitiveDescriptor()->getConfig().outConfs.size();
+    for (size_t i = 0; i < num_ports; i++) {
+        auto childEdge = node->getChildEdgeAt(i);
+
+        std::string file_name = nodeName;
+        if (count != -1)
+            file_name += "_iter" + std::to_string(count);
+        file_name += "_out" + std::to_string(i) + ".ieb";
+        if (file_name.size() > 240)
+            file_name = file_name.substr(file_name.size() - 240);
+
+        auto dump_file = dumpDirName + "/#" + exec_order + "_" + file_name;
+        std::cout << "Dump after:  " << dump_file << std::endl;
+
+        TensorDesc desc = childEdge->getDesc();
+        if (desc.getPrecision() == Precision::BIN)
+            continue;
+
+        BlobDumper dumper(childEdge->getBlob());
+        if (node->ext_scales)
+            dumper.withScales(node->ext_scales);
+
+        if (shouldDumpAsText)
+            dumper.dumpAsTxt(dump_file);
+        else
+            dumper.dump(dump_file);
+    }
+}
+
+void NodeDumper::dumpInternalBlobs(const MKLDNNNodePtr& node) const {
+    for (size_t i = 0; i < node->internalBlobs.size(); i++) {
+        const auto& blb = node->internalBlobs[i];
+        auto dump_file = dumpDirName + "/#" + std::to_string(node->getExecIndex()) + "_" + node->getName() + "_blb" + std::to_string(i) + ".ieb";
+        TensorDesc desc = blb->getTensorDesc();
+        if (desc.getPrecision() == Precision::BIN)
+            continue;
+        BlobDumper dumper(blb);
+        if (shouldDumpAsText)
+            dumper.dumpAsTxt(dump_file);
+        else
+            dumper.dump(dump_file);
+    }
+}
+
+bool NodeDumper::shouldBeDumped(const MKLDNNNodePtr& node) const {
+    bool shouldBeDumped = false;
+    const std::string& filterById = dumpFilters[FILTER::BY_EXEC_ID];
+
+    if (!filterById.empty()) {                              // filter by exec id env set
+        std::stringstream ss(filterById);
+        int id;
+        while (ss >> id) {
+            if (node->getExecIndex() == id) // exec id matches
+                shouldBeDumped = true;
+        }
+
+        if (!shouldBeDumped)
+            return false;
+    }
+
+    const std::string& filterByType = dumpFilters[FILTER::BY_TYPE];
+
+    if (!filterByType.empty()) {                           // filter by type env set
+        if (NameFromType(node->getType()) != filterByType) // type does not match
+            return false;
+        else
+            shouldBeDumped = true;
+    }
```

Comment:
It would be more elegant not to use the else clause, just initialize `shouldBeDumped` with true and return immediately if the condition is not met.

Reply:
The problem is that we do not want to dump by default, so if none of the filters were matched we want to return false

Reply:
I agree this implementation is very ugly, glad to receive any recommendations how to make it better.

Reply:
For example we can make `dumpFilters` to be a dynamic object, (for example hash table so we can convert the enum to enum class, or vector, etc) and add filters there. Then just check this `dumpFilters` container and return false if empty or make all the further checks if not.

Reply:
Done